Jenkinsfile

```diff
@@ -0,0 +1,630 @@
+  def branchNameNoPeriods = env.BRANCH_NAME.replaceAll('\\.', '_')
+  def durationMetric = "okr.ci.nodejs.${env.DRIVER_METRIC_TYPE}.${buildType}.${branchNameNoPeriods} ${durationSec} ${nowSec}"
+
+  timeout(time: 1, unit: 'MINUTES') {
+    withCredentials([string(credentialsId: 'lab-grafana-address', variable: 'LAB_GRAFANA_ADDRESS'),
+                     string(credentialsId: 'lab-grafana-port', variable: 'LAB_GRAFANA_PORT')]) {
+      withEnv(["DURATION_METRIC=${durationMetric}"]) {
+        sh label: 'Send runtime metrics to labgrafana', script: '''#!/bin/bash -lex
+          echo "${DURATION_METRIC}" | nc -q 5 ${LAB_GRAFANA_ADDRESS} ${LAB_GRAFANA_PORT}
+        '''
+      }
+    }
+  }
+}
+
+def describePerCommitStage() {
+  script {
+    currentBuild.displayName = "Per-Commit"
+    currentBuild.description = '''Per-Commit build and testing'''
+  }
+}
+
+def describeScheduledTestingStage() {
+  script {
+    def type = params.CI_SCHEDULE.toLowerCase().capitalize()
+    currentBuild.displayName = "${type} schedule"
+    currentBuild.description = "${type} scheduled build and testing of all supported Apache Cassandra and DataStax " +
+            "Enterprise against multiple Node.js runtimes"
+  }
+}
+
+def describeAdhocTestingStage() {
+  script {
+    def serverType = params.ADHOC_BUILD_AND_EXECUTE_TESTS_SERVER_VERSION.split('-')[0]
+    def serverDisplayName = 'Apache Cassandra'
+    def serverVersion = " v${serverType}"
+    if (serverType == 'ALL') {
+      serverDisplayName = "all ${serverDisplayName} and DataStax Enterprise server versions"
+      serverVersion = ''
+    } else {
+      try {
+        serverVersion = " v${env.ADHOC_BUILD_AND_EXECUTE_TESTS_SERVER_VERSION.split('-')[1]}"
+      } catch (e) {
+        ;; // no-op
+      }
+      if (serverType == 'ddac') {
+        serverDisplayName = "DataStax Distribution of ${serverDisplayName}"
+      } else if (serverType == 'dse') {
+        serverDisplayName = 'DataStax Enterprise'
+      }
+    }
+
+    def nodeJsVersionInformation = "Node.js v${params.ADHOC_BUILD_AND_EXECUTE_TESTS_NODEJS_VERSION}"
+    if (params.ADHOC_BUILD_AND_EXECUTE_TESTS_NODEJS_VERSION == 'ALL') {
+      nodeJsVersionInformation = 'all Node.js versions'
+    }
+
+    def examplesInformation = ''
+    if (ADHOC_BUILD_AND_EXECUTE_TESTS_EXECUTE_EXAMPLES) {
+      examplesInformation = ' with examples'
+    }
+
+    currentBuild.displayName = "${params.ADHOC_BUILD_AND_EXECUTE_TESTS_SERVER_VERSION} against ${nodeJsVersionInformation}"
+    currentBuild.description = "Testing ${serverDisplayName} ${serverVersion} against ${nodeJsVersionInformation}${examplesInformation}"
+  }
+}
+
+pipeline {
+  agent none
+
+  // Global pipeline timeout
+  options {
+    timeout(time: 10, unit: 'HOURS')
+    buildDiscarder(logRotator(artifactNumToKeepStr: '10', // Keep only the last 10 artifacts
+                              numToKeepStr: '50'))        // Keep only the last 50 build records
+  }
```

Comment:
on the old jenkins we only allowed one concurrent build per branch, if you want the same behavior you need to add `disableConcurrentBuilds()` here (e.g. https://github.com/datastax/csharp-driver/blob/master/Jenkinsfile#L396)

Reply:
Done!